### PR TITLE
More quad input hatches

### PR DESCRIPTION
**Pull request by TheEpicGamer274**
More Quad Inputs Umv, uxv and the humongous version because some people have been asking for it. New class because I didn't believe it was possible in the current class for Quad Input Hatches, recipes will be done in coremod after this pr.

### CODE REVIEW

File: src/main/java/gregtech/api/metatileentity/implementations/GT_MetaTileEntity_Hatch_QuadrupleHumongous.java [hunk 1]
```diff
@@ -0,0 +1,299 @@
...
+import com.gtnewhorizons.modularui.common.fluid.FluidStackTank;
+import com.gtnewhorizons.modularui.common.widget.FluidSlotWidget;
+
+import gregtech.api.interfaces.ITexture;
+import gregtech.api.interfaces.modularui.IAddUIWidgets;
+import gregtech.api.interfaces.tileentity.IGregTechTileEntity;
+import gregtech.api.metatileentity.MetaTileEntity;
+import gregtech.api.render.TextureFactory;
+
+public class GT_MetaTileEntity_Hatch_QuadrupleHumongous extends GT_MetaTileEntity_Hatch_Input implements IAddUIWidgets {
```
Comment (miozune) [inline src/main/java/gregtech/api/metatileentity/implementations/GT_MetaTileEntity_Hatch_QuadrupleHumongous.java:23]: This class is almost identical to `GT_MetaTileEntity_Hatch_MultiInput`. Add method `getCapacityPerTank` or smth to `GT_MetaTileEntity_Hatch_Input` and make this class extend `GT_MetaTileEntity_Hatch_MultiInput`.

**Review by miozune — CHANGES_REQUESTED** [review]

File: src/main/java/gregtech/api/metatileentity/implementations/GT_MetaTileEntity_Hatch_Input.java [hunk 2]
```diff
@@ -45,6 +45,10 @@ public GT_MetaTileEntity_Hatch_Input(int aID, int aSlot, String aName, String aN
                 "Can hold " + aSlot + " types of fluid." });
```
Comment (miozune) [inline src/main/java/gregtech/api/metatileentity/implementations/GT_MetaTileEntity_Hatch_Input.java:45]: Ugly but working solution: 
```java
        super(
            aID,
            aName,
            aNameRegional,
            aTier,
            aSlot,
            new String[] { "Fluid Input for Multiblocks", "", "Can hold " + aSlot + " types of fluid." });
        mDescriptionArray[1] = "Capacity: " + GT_Utility.formatNumbers(getCapacityPerTank(aTier, aSlot)) + "L";
```

File: src/main/java/gregtech/api/metatileentity/implementations/GT_MetaTileEntity_Hatch_QuadrupleHumongous.java [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+
+public class GT_MetaTileEntity_Hatch_QuadrupleHumongous extends GT_MetaTileEntity_Hatch_MultiInput {
+
+    public final int mCapacityPer;
+    private final int Capacity = 500000000;
+
+    public GT_MetaTileEntity_Hatch_QuadrupleHumongous(int aID, int aSlot, String aName, String aNameRegional) {
+        super(aID, aSlot, aName, aNameRegional, 0);
+        mCapacityPer = Capacity;
+        this.mDescriptionArray[1] = "Capacity: 500,000,000L";
```
Comment (miozune) [inline src/main/java/gregtech/api/metatileentity/implementations/GT_MetaTileEntity_Hatch_QuadrupleHumongous.java:15]: Then this is not needed

File: src/main/java/gregtech/api/metatileentity/implementations/GT_MetaTileEntity_Hatch_QuadrupleHumongous.java [hunk 4]
```diff
@@ -0,0 +1,29 @@
+package gregtech.api.metatileentity.implementations;
+
+import gregtech.api.interfaces.ITexture;
+import gregtech.api.interfaces.tileentity.IGregTechTileEntity;
+import gregtech.api.metatileentity.MetaTileEntity;
+
+public class GT_MetaTileEntity_Hatch_QuadrupleHumongous extends GT_MetaTileEntity_Hatch_MultiInput {
+
+    public final int mCapacityPer;
+    private final int Capacity = 500000000;
```
Comment (miozune) [inline src/main/java/gregtech/api/metatileentity/implementations/GT_MetaTileEntity_Hatch_QuadrupleHumongous.java:10]: Unnecessary

File: src/main/java/gregtech/api/metatileentity/implementations/GT_MetaTileEntity_Hatch_QuadrupleHumongous.java [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+        ITexture[][][] aTextures) {
+        super(aName, aSlot, aTier, aDescription, aTextures);
+        mCapacityPer = Capacity;
+    }
+
+    @Override
+    public MetaTileEntity newMetaEntity(IGregTechTileEntity aTileEntity) {
+        return new GT_MetaTileEntity_Hatch_QuadrupleHumongous(mName, getMaxType(), mTier, mDescriptionArray, mTextures);
+
+    }
```
Comment (miozune) [inline src/main/java/gregtech/api/metatileentity/implementations/GT_MetaTileEntity_Hatch_QuadrupleHumongous.java:28]: Add
```java
@Override
public int getCapacityPerTank(int aTier, int aSlot) {
    return 500000000;
}
```

**Review by miozune — COMMENTED** [review]
Replace `mCapacityPer = 8000 * (1 << aTier) / aSlot;` in `GT_MetaTileEntity_Hatch_MultiInput` with `mCapacityPer = getCapacityPerTank(aTier, aSlot);`

File: src/main/java/gregtech/api/metatileentity/implementations/GT_MetaTileEntity_Hatch_QuadrupleHumongous.java [hunk 6]
```diff
@@ -0,0 +1,28 @@
...
+
+    @Override
+    public int getCapacityPerTank(int aTier, int aSlot) {
+        return 500000000;
+    }
+
+    @Override
+    public MetaTileEntity newMetaEntity(IGregTechTileEntity aTileEntity) {
+        return new GT_MetaTileEntity_Hatch_QuadrupleHumongous(mName, getMaxType(), mTier, mDescriptionArray, mTextures);
+
```
Comment (miozune) [inline src/main/java/gregtech/api/metatileentity/implementations/GT_MetaTileEntity_Hatch_QuadrupleHumongous.java:26]: nit: remove extra line

File: src/main/java/gregtech/api/metatileentity/implementations/GT_MetaTileEntity_Hatch_QuadrupleHumongous.java [hunk 7]
```diff
@@ -0,0 +1,28 @@
...
+    }
+
+    public GT_MetaTileEntity_Hatch_QuadrupleHumongous(String aName, int aSlot, int aTier, String[] aDescription,
+        ITexture[][][] aTextures) {
+        super(aName, aSlot, aTier, aDescription, aTextures);
+    }
+
+    @Override
+    public int getCapacityPerTank(int aTier, int aSlot) {
+        return 500000000;
```
Comment (minecraft7771) [inline src/main/java/gregtech/api/metatileentity/implementations/GT_MetaTileEntity_Hatch_QuadrupleHumongous.java:20]:
```suggestion
        return 500_000_000;
```

**Review by miozune — APPROVED** [review]

**Review by minecraft7771 — APPROVED** [review]

**Pull request merged**